herbert_core/applications/mfit_funcs/slow_func1d.m

```diff
@@ -0,0 +1,44 @@
+function y = slow_func1d (x, p, funchandle, nslow)
+% Evaluate a function but make it much slower 
+%
+%   >> y = slow_func1d (x, p, funchandle, nslow)
+%
+% Input:
+% =======
+%   x           Array of values at which to evaluate function along the first
+%               dimension
+%
+%   p           Vector of parameters needed by the function funchandle:
+%               E.G. if a two dimensional Gaussian (see gauss)
+%                   p = [height, centre, st_deviation]
+%
+%   funchandle  Handle to function to evaluate e.g. @gauss
+%               The function can be any function that evaluates in
+%               two dimensions with the format expected by multifit
+%
+% Optionally:
+%   nslow       Number of times to run the time_waster function, which
+%               alters each calculated value from the function call by a
+%               factor <= 10^-13 (regardless of the value of nslow.
+%               Each value of nslow takes about the time of 25 
+%               exponentiations.
+%                   nslow >=0
+%               Default if not given: nslow = 1
+%
+% Output:
+% ========
+%   y           Array of calculated y-axis values
+
+if nargin==3
```

Comment:
Also, rather than using the less clear `nargin` approach it might be better to use an inputParser. Cost is negligible as this is time-wasting. 
```matlab
ip = inputParser();
addRequired(ip, 'x', @iscell);
addRequired(ip, 'p', @(x) validateattributes(x, {'numeric'}, {'vector'}}));
addRequired(ip, 'funchandle', @(x) isa(x, 'function_handle'));
addOptional(ip, 'nslow', 1, @(x) validateattributes(x, {'numeric'}, {'scalar', 'integer', 'nonnegative'}));
parse(ip, x, p, funchandle, nslow);
[x, p, funchandle, nslow] = struct2cell(ip.Results);
```

Reply:
**N.B.** I have not checked this works, it is just an example.

Reply:
Just never used it before, and wanted to get something in quick...